File: data-automation-project/backend/data-automation-project/src/main/java/com/studies/lvb/data_automation_project/controller/UserController.java
package com.studies.lvb.data_automation_project.controller;

import com.studies.lvb.data_automation_project.dto.UserDTO;
import com.studies.lvb.data_automation_project.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/user")
public class UserController {

    @Autowired
    private UserService userService;

    @PostMapping("/register")
    public ResponseEntity<UserDTO> createUser(@RequestBody UserDTO userDTO) {
        userDTO = userService.createUser(userDTO);
        return ResponseEntity.ok(userDTO);
    }
}

File: BE-Dev-Courses/Java/Spring JPA, RESTFul, Security, JWT/point-of-sale/point-of-sale-api/jwt/src/main/kotlin/com/lvb/studys/kotlin/api/pointofsaleapi/jwt/controller/UserController.kt
package com.lvb.studys.kotlin.api.pointofsaleapi.jwt.controller

import com.lvb.studys.kotlin.api.pointofsaleapi.jwt.dto.ResponseDTO
import com.lvb.studys.kotlin.api.pointofsaleapi.jwt.dto.UserDTO
import com.lvb.studys.kotlin.api.pointofsaleapi.jwt.exceptions.InvalidOperationException
import com.lvb.studys.kotlin.api.pointofsaleapi.jwt.exceptions.NoItemException
import com.lvb.studys.kotlin.api.pointofsaleapi.jwt.service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.dao.EmptyResultDataAccessException
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/user")
class UserController @Autowired constructor(
    private val userService: UserService
) {

    @GetMapping
    fun getAll(): ResponseEntity<List<UserDTO>> {
        return ResponseEntity.status(HttpStatus.OK).body(userService.getAllUsers())
    }

    @PostMapping
    fun saveUser(@RequestBody user: UserDTO): ResponseEntity<Any> {
        user.isEnable = true
        userService.saveUser(user)
        return try {
            ResponseEntity(HttpStatus.CREATED)
        } catch (e: Exception) {
            ResponseEntity(e.message, HttpStatus.CREATED)
        }
    }

    @PutMapping
    fun updateUser(@RequestBody user : UserDTO) : ResponseEntity<Any> {
        if (user.id == null) {
            throw InvalidOperationException("User's ID need to be informed")
        }

        return try {
            ResponseEntity.status(HttpStatus.OK).body(userService.updateUser(user))
        } catch (e: NoItemException) {
            ResponseEntity(ResponseDTO(e.message ?: ""), HttpStatus.BAD_REQUEST)
        } catch (e: Exception) {
            ResponseEntity(ResponseDTO(e.message ?: ""), HttpStatus.INTERNAL_SERVER_ERROR)
        }

    }

    @DeleteMapping("/{id}")
    fun deleteUser(@PathVariable("id") id: Long): ResponseEntity<Any> {
        return try {
            userService.deleteUser(id)
            ResponseEntity.status(HttpStatus.OK).body(ResponseDTO("User deleted successfully"))
        } catch (e: EmptyResultDataAccessException) {
            ResponseEntity(ResponseDTO("User not found"), HttpStatus.INTERNAL_SERVER_ERROR)
        } catch (e: Exception) {
            ResponseEntity(e.message, HttpStatus.INTERNAL_SERVER_ERROR)
        }
    }
}